dffml/base.py

```diff
@@ -235,7 +259,61 @@ def config(cls):
     """
     Decorator to create a dataclass
     """
-    datacls = dataclasses.dataclass(eq=True, init=True)(cls)
+    class_list = []
+    newdict = {}
+    defaultdict = {}
+    if len(inspect.getmro(cls)) > 2:
+        class_list = list(inspect.getmro(cls)[:-1])
+    else:
+        class_list = [cls]
+    for clss in class_list:
+        cls_annotations = clss.__dict__.get("__annotations__", {})
+        cls_dict = clss.__dict__
+        for var in cls_annotations:
+            if not dataclasses.is_dataclass(clss):
```

Comment:
Since the body of the each for loop is the same, we should make the body a function